File: crates/domain-scan-core/tests/fixtures/scala/traits.scala
package com.example

trait UserRepository {
  def findById(id: String): Option[User]
  def findAll(): List[User]
  def save(user: User): Unit
  def delete(id: String): Unit
}

trait Closeable {
  def close(): Unit
}

trait EventHandler[T] extends Closeable {
  def handle(event: T): Unit
  def canHandle(eventType: String): Boolean
}

private[example] trait InternalCache {
  def get(key: String): Option[Any]
}

File: crates/domain-scan-core/tests/fixtures/rust/traits.rs
pub trait EventHandler {
    fn handle(&self, event: Event) -> Result<(), Error>;
    fn name(&self) -> &str;
}

pub trait Repository<T> {
    fn find_by_id(&self, id: u64) -> Option<T>;
    fn save(&mut self, item: T) -> Result<(), Error>;
    fn delete(&self, id: u64) -> Result<(), Error>;
}

pub trait Serializable: Clone + Send {
    fn serialize(&self) -> Vec<u8>;
    fn deserialize(data: &[u8]) -> Self;
}

trait PrivateTrait {
    fn internal_method(&self);
}
